src/pages/invoices/common/hooks/useTaskColumns.ts

```diff
@@ -16,41 +16,127 @@ export function useTaskColumns() {
   const company = useCurrentCompany();
   const [columns, setColumns] = useState<string[]>([]);
 
+  const getColumnIndex = (
+    columnKey: string,
+    pdfColumns: string[],
+    currentColumns: string[]
+  ) => {
+    const columnIndex = pdfColumns.findIndex(
+      (variable) => variable === columnKey
+    );
+
+    const adjustedColumnIndex =
+      columnIndex > -1 ? columnIndex : currentColumns.length;
+
+    return adjustedColumnIndex;
+  };
+
   useEffect(() => {
-    // We need to clone the product columns to local object,
-    // because by default it's frozen.
-    let variables: string[] =
```

Comment:
The same comments go as before. No need to remove comments, update them if needed & don't use splice unless it's absolutely needed.

Reply:
Sure, resolved!